### PR TITLE
Speed up Segment write-out of ExpressionEvaluator

### PR DESCRIPTION
**Pull request by mrzzzrm**
`ExpressionEvaluator::evaluate_expression_to_segment()` used to materialize the result of an expression into a concurrent vector, which is insanely slow. Now it materializes into vectors first and passes them to the concurrent vector's constructor, which is faster. Mostly does good for TPC-H 1.

```
+-----------+-----------------+------+-----------------+------+--------+---------------------------------+
| Benchmark | prev. iter/s    | runs | new iter/s      | runs | change | p-value (significant if <0.001) |
+-----------+-----------------+------+-----------------+------+--------+---------------------------------+
| TPC-H 1   | 0.212431043386  | 13   | 0.24973641336   | 16   | +18%   |                          0.0000 |
| TPC-H 2   | 3.24405765533   | 20   | 3.23968243599   | 20   | -0%    |     (run time too short) 0.9421 |
| TPC-H 3   | 2.87702393532   | 20   | 2.94130849838   | 20   | +2%    |     (run time too short) 0.1071 |
| TPC-H 4   | 0.843577206135  | 20   | 0.842742919922  | 20   | -0%    |     (run time too short) 0.8548 |
| TPC-H 5   | 2.26284217834   | 20   | 2.24027323723   | 20   | -1%    |     (run time too short) 0.4375 |
| TPC-H 6   | 8.26083564758   | 20   | 8.36928749084   | 20   | +1%    |     (run time too short) 0.4322 |
| TPC-H 7   | 0.640364646912  | 20   | 0.644297242165  | 20   | +1%    |     (run time too short) 0.1034 |
| TPC-H 8   | 2.80517339706   | 20   | 2.74800443649   | 20   | -2%    |     (run time too short) 0.0519 |
| TPC-H 9   | 0.733802258968  | 20   | 0.74705606699   | 20   | +2%    |     (run time too short) 0.0051 |
| TPC-H 10  | 1.22619497776   | 20   | 1.33375298977   | 20   | +9%    |     (run time too short) 0.0000 |
| TPC-H 11  | 17.3254833221   | 20   | 17.8998374939   | 20   | +3%    |     (run time too short) 0.1954 |
| TPC-H 12  | 3.0418817997    | 20   | 3.07603883743   | 20   | +1%    |     (run time too short) 0.6085 |
| TPC-H 13  | 1.07812929153   | 20   | 1.074213624     | 20   | -0%    |     (run time too short) 0.8520 |
| TPC-H 14  | 7.34988975525   | 20   | 7.44479560852   | 20   | +1%    |     (run time too short) 0.7598 |
| TPC-H 15  | 4.70872735977   | 20   | 4.41249704361   | 20   | -6%    |     (run time too short) 0.0000 |
| TPC-H 16  | 3.64220833778   | 20   | 3.67610192299   | 20   | +1%    |     (run time too short) 0.4054 |
| TPC-H 17  | 0.0449181050062 | 3    | 0.0489784963429 | 3    | +9%    |        (not enough runs) 0.0682 |
| TPC-H 18  | 0.849766135216  | 20   | 0.832519412041  | 20   | -2%    |     (run time too short) 0.0344 |
| TPC-H 19  | 2.64768767357   | 20   | 2.59783291817   | 20   | -2%    |     (run time too short) 0.0345 |
| TPC-H 20  | 0.0272622555494 | 2    | 0.0274009481072 | 2    | +1%    |        (not enough runs) 0.5994 |
| TPC-H 21  | 0.0481206774712 | 3    | 0.0462750829756 | 3    | -4%    |        (not enough runs) 0.3193 |
| TPC-H 22  | 3.9603536129    | 20   | 4.0112953186    | 20   | +1%    |     (run time too short) 0.3243 |
| average   |                 |      |                 |      | +1%    |                                 |
+-----------+-----------------+------+-----------------+------+--------+---------------------------------+
```

### CODE REVIEW

File: src/bin/CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,7 @@ add_executable(
 target_link_libraries(
     hyrisePlayground
     hyrise
+    hyriseBenchmarkLib
```
Comment (mrzzzrm) [inline src/bin/CMakeLists.txt]: on purpose.

**Comment by Bouncner** [comment]
I don’t see where you pass the values to the concurrent vector in the changes. Can you help me here?

Is this additional work required because the ValueSegment is still using TBB?
If so, add a TODO to change that once we get rid of the tbb vectors?

**Comment by mrzzzrm** [comment]
> I don’t see where you pass the values to the concurrent vector in the changes

The Value Segment has additional constructors which take `std::vector` - so it just works magically. For the same reason, no TODO for when we remove TBB from the ValueSegment is needed.

**Comment by Bouncner** [comment]
Nice catch.

**Comment by mrks** [comment]
Argh, forgot about the commit msg

**Pull request merged**